src/lib/cache/AccessibilityCloudImageCache.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import URLDataCache from './URLDataCache';
+import type { AccessibilityCloudImages } from '../Feature';
+import config from '../config';
+
+export const InvalidCaptchaReason = "invalid-captcha";
+export const UnknownReason = "unknown";
+
+export default class AccessibilityCloudImageCache extends URLDataCache<AccessibilityCloudImages> {
+  
+  getPhotosForFeature(featureId: string): Promise<AccessibilityCloudImages | null> {
+    return this.getImage("place", featureId);
+  }
+
+  getImage(context: string, objectId: string): Promise<AccessibilityCloudImages | null> {
+    return this.getData(`${config.accessibilityCloudBaseUrl}/images.json?context=${context}&objectId=${objectId}&appToken=${config.accessibilityCloudAppToken}`);
+  }
+
+  uploadPhotoForFeature(featureId: string, images: FileList, captchaSolution: string): Promise<boolean> {
+    const image = images[0];
+
+    const uploadPromise = new Promise((resolve, reject) => {
+      this.constructor.fetch(
+        `${config.accessibilityCloudBaseUrl}/image-upload?placeId=${featureId}&captcha=${captchaSolution}&appToken=${config.accessibilityCloudAppToken}`, 
+        {
+          method: "POST",
+          headers: {
+            'Accept': 'application/json',
+            'Content-Type': image.type,
+          },
+          body: image,
+        }
+      ).then((response: Response) => {
+        if (response.ok) {
+          resolve(true);
+        } else {
+          response.json()
+            .then((json) => {
+              const reason = json.error.reason;
+              if (reason === "No captcha found.") {
+                reject(InvalidCaptchaReason);
+              } else {
+                reject(UnknownReason);
+              }
+            }).catch(reject);
+        }
+      }).catch(reject);
+    });
+
+    return uploadPromise;
+  }
+  
+  getCaptcha() : Promise<string> {
+    let promise = this.captchaRequest;
+    if (promise) return promise;
+
+    const url = `${config.accessibilityCloudBaseUrl}/captcha.svg?appToken=${config.accessibilityCloudAppToken}`
+    promise = new Promise((resolve, reject) => {
+      if (this.hasValidCaptcha() && this.lastCaptcha) {
+        resolve(this.lastCaptcha);
+      } else {
+        console.log("requesting new captcha");
+        this.lastCaptcha = null;
+        return this.constructor.fetch(url).then((response: Response) => {
+          if (response.ok) {
+            const expires = response.headers.get('expires');
```

Comment:
Does this already work with the deployed remote AC API? Anything we have to keep in mind regarding CORS?

Reply:
I just tried against my local ac, but works fine. I had to add some cors stuff for localhost:4000, but otherwise its cool.